src/routes/docs/tutorials/pagination/step-7/+page.markdoc

```diff
@@ -0,0 +1,265 @@
+
+```js
+import { createContext, useContext, useEffect, useState } from "react";
+import { databases } from "../appwrite";
+import { Query } from "appwrite";
+
+const TODOS_DATABASE_ID = "[YOUR_DATABASE_ID]";
+const TODOS_COLLECTION_ID = "[YOUR_COLLECTION_ID]";
+
+const TodosContext = createContext();
+
+export function useTodos() {
+  return useContext(TodosContext);
+}
+
+export function TodosProvider(props) {
+  const [todos, setTodos] = useState([]);
+  const [offset, setOffset] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [counts, setCounts] = useState({
+    totalPages: 500,
+    totalResults: 1000,
+  });
+  const pageLimit = 4;
+  const hasNext = counts.totalPages > currentPage;
+
+  async function loadNext() {
+    if (hasNext) {
+      setOffset(offset + pageLimit);
+    }
+  }
+
+  async function loadPrev() {
+    if (offset >= pageLimit) {
+      setOffset(offset - pageLimit);
+    }
+  }
+
+  async function loadPage(pageNumber) {
+    setOffset((pageNumber - 1) * pageLimit);
+  }
+
+  async function init() {
+    try {
+      const response = await databases.listDocuments(
+        TODOS_DATABASE_ID,
+        TODOS_COLLECTION_ID,
+        [
+          Query.orderAsc("$createdAt"),
+          Query.limit(pageLimit),
+          Query.offset(offset),
+        ],
+      );
+
+      setTodos(response.documents);
+      setCurrentPage(Math.floor(offset / pageLimit) + 1);
+      setCounts({
+        totalPages: response.total / pageLimit,
+        totalResults: response.total,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  useEffect(() => {
+    init();
+  }, [offset]);
+
+  return (
+    <TodosContext.Provider
+      value={{
+        current: todos,
+        loadNext,
+        loadPrev,
+        loadPage,
+        currentPage,
+        totalPages: counts.totalPages,
+        hasNext,
+        hasPrev: currentPage > 1,
+      }}>
+      {props.children}
+    </TodosContext.Provider>
+  );
+}
+```
+In this updated context:
+ * We're using `Query.limit()` to set the number of tasks we want to fetch per page.
+ * We're using `Query.offset()` to set the starting point for fetching tasks based on the current page.
+ * We've added `loadNextPage`, `loadPrevPage`, and `loadPage` function to handle pagination.
+
+ Now, you can use this context in your components to display todo items and handle pagination using Appwrite's offset pagination methods.
+
+# Update the Todos page {% #update-the-todoss-page %}
+
+Now, let's use our Pagination component in the Todos page.
+{% tabs %}
+{% tabsitem #Basic-pagination title="Basic Pagination" %}
+Update the Todos page in `src/pages/Todos.jsx`:
+```js
+import BasicPagination from '../components/BasicPagination';
+import {useTodos} from "../lib/context/todos";
+export function TodosWithBasicPagination() {
```

Comment:
One of the issues this will have is that we are expecting this to export Todo.

The name here isn't quite right :D

Reply:
I was trying to keep all the TODOs with different pagination methods separated in a demo. I guess I forgot a couple.